aderyn_core/src/detect/high/unchecked_send.rs

```diff
@@ -0,0 +1,106 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::ast::{ASTNode, NodeID, NodeType};
+
+use crate::capture;
+use crate::context::browser::GetImmediateParent;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct UncheckedSendDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for UncheckedSendDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for member_access in context.member_accesses() {
+            if member_access.member_name == "send"
+                && member_access
+                    .expression
+                    .type_descriptions()
+                    .is_some_and(|type_desc| {
+                        type_desc
+                            .type_string
+                            .as_ref()
+                            .is_some_and(|type_string| type_string.starts_with("address"))
+                    })
+            {
+                if let Some(ASTNode::FunctionCall(func_call)) = member_access.parent(context) {
+                    if func_call
+                        .parent(context)
+                        .is_some_and(|node| node.node_type() == NodeType::Block)
+                    {
+                        capture!(self, context, func_call);
+                    }
+                }
+            }
+        }
+
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::High
+    }
+
+    fn title(&self) -> String {
+        String::from("Unchecked `bool success` value for send call.")
+    }
+
+    fn description(&self) -> String {
+        String::from("`address(payable?).send(address)` returns a boolean that must be used to verify if the transaction was successful")
```

Comment:
We want to teach devs the reasons behind each detector, so the description should include:
* Why this is dangerous
* What the mitigation is

Reply:
Understood

Reply:
✅

Reply:
Beautiful description @TilakMaddy ! 😍